### PR TITLE
refactor(schematics): rename ast json methods more explicitly

### PR DESCRIPTION
**Pull request by FrozenPandaz**
Renamed the ast Json methods to have InTree at the end.

Also added docs to describe intended usage

### CODE REVIEW

File: packages/bazel/src/utils/fileutils.ts [hunk 1]
```diff
@@ -2,6 +2,14 @@ import * as fs from 'fs';
 import { Tree } from '@angular-devkit/schematics';
 import * as path from 'path';
 
+/**
+ * This method is specifically for updating a JSON using the filesystem
```
Comment (JamesHenry) [inline packages/bazel/src/utils/fileutils.ts]: missing the word "file" I think

File: packages/bazel/src/utils/fileutils.ts [hunk 2]
```diff
@@ -30,6 +38,13 @@ export function serializeJson(json: any): string {
   return `${JSON.stringify(json, null, 2)}\n`;
 }
 
+/**
+ * This method is specifically for reading a JSON from the filesystem
```
Comment (JamesHenry) [inline packages/bazel/src/utils/fileutils.ts]: missing "file" again

File: packages/schematics/src/utils/fileutils.ts [hunk 3]
```diff
@@ -2,10 +2,14 @@ import * as fs from 'fs';
 import { Tree } from '@angular-devkit/schematics';
 import * as path from 'path';
 
-export function readJsonFile(path: string) {
-  return JSON.parse(fs.readFileSync(path, 'utf-8'));
-}
-
+/**
+ * This method is specifically for updating a JSON using the filesystem
```
Comment (JamesHenry) [inline packages/schematics/src/utils/fileutils.ts]: "a JSON" again

File: packages/schematics/src/utils/fileutils.ts [hunk 4]
```diff
@@ -2,10 +2,14 @@ import * as fs from 'fs';
...
+ *
+ * @remarks
+ * If you are looking to update a JSON in a tree, look for ./ast-utils#updateJsonInTree
```
Comment (JamesHenry) [inline packages/schematics/src/utils/fileutils.ts]: "a JSON" again

File: packages/schematics/src/utils/fileutils.ts [hunk 5]
```diff
@@ -34,6 +38,17 @@ export function serializeJson(json: any): string {
   return `${JSON.stringify(json, null, 2)}\n`;
 }
 
+/**
+ * This method is specifically for reading a JSON from the filesystem
```
Comment (JamesHenry) [inline packages/schematics/src/utils/fileutils.ts]: "a JSON" again

File: packages/schematics/src/utils/fileutils.ts [hunk 6]
```diff
@@ -34,6 +38,17 @@ export function serializeJson(json: any): string {
...
+ *
+ * @remarks
+ * If you are looking to read a JSON in a Tree, use ./ast-utils#readJsonInTree
```
Comment (JamesHenry) [inline packages/schematics/src/utils/fileutils.ts]: "a JSON" again

**Comment by JamesHenry** [comment]
Thanks @FrozenPandaz! Those changes help improve the readability a lot.

Just found a lot of the same "a JSON" typos in comments, nothing major.

**Review by JamesHenry — APPROVED** [review]

**Comment by ThomasBurleson** [comment]
This will impact PR #359... and should be rebased after 359 has been merged.

**Comment by FrozenPandaz** [comment]
I'm fine with making this change after all the other changes have been merged. It's low effort to resolve those merge conflicts

**Pull request merged**